frontend/src/components/common/snackbar/SnackbarContext.tsx

```diff
@@ -0,0 +1,70 @@
+'use client';
+import type { ReactNode } from 'react';
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import Snackbar from '@mui/material/Snackbar';
+import Alert from '@mui/material/Alert';
+
+interface SnackbarContextType {
+  openSnackbar: (message: string, severity?: 'info' | 'warning') => void;
+  closeSnackbar: () => void;
+}
+
+const SnackbarContext = createContext<SnackbarContextType | undefined>(undefined);
+
+interface SnackbarProviderProps {
+  children: ReactNode;
+}
+
+export const SnackbarProvider: React.FC<SnackbarProviderProps> = ({ children }) => {
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState<'info' | 'warning' | undefined>(
```

Comment:
Waht about adding type for severity include `error` too. For example,
```
type Severity = 'info' | 'warning' | 'error' | undefined;
```

And use this type for openSnackbar and snackbarSeverity state